test/TestResourcesAndFunctions.cpp

```diff
@@ -0,0 +1,259 @@
+    );
+}
+
+/**
+ * No test for this, tested within other tests.
+ */
+bool TestResourcesAndFunctions::createConcatenatedFile(const path &file, const path &result, int repetitions) {
+            CHECK(repetitions > 0);
+    bool res = true;
+    /**
+     * First time, create a fresh file.
+     */
+    res &= runCommand(
+            format(R"(cat "%s" > "%s")",
+                   file.string().c_str(),
+                   result.string().c_str()
+            )
+    );
+    /**
+     * Second to n-time append gz.
+     */
+    for (int i = 1; i < repetitions; i++) {
+        res &= runCommand(
+                format(R"(cat "%s" >> "%s")",
+                       file.string().c_str(),
+                       result.string().c_str()
+                )
+        );
+    }
+    return res;
+}
+
+vector<string> TestResourcesAndFunctions::readLinesOfFile(const path &file) {
+    ifstream strm(file);
+    vector<string> decompressedSourceContent;
+    string line;
+    while (std::getline(strm, line)) {
+        decompressedSourceContent.emplace_back(line);
+    }
+    return decompressedSourceContent;
+}
+
+string TestResourcesAndFunctions::readFile(const path &file) {
+    std::ifstream t(file);
+    std::string str;
+
+    t.seekg(0, std::ios::end);
+    str.reserve(t.tellg());// TODO
+    t.seekg(0, std::ios::beg);
+
+    str.assign((std::istreambuf_iterator<char>(t)), std::istreambuf_iterator<char>());
+    return str;
+}
+
+bool TestResourcesAndFunctions::compareVectorContent(const vector<string> &reference, const vector<string> &actual,
+                                                     uint32_t referenceOffset) {
+    int64_t firstDiff{-1};
+    for (u_int64_t i = 0; i < std::min(reference.size() - referenceOffset, actual.size()); i++) {
+        if (reference[i + referenceOffset] != actual[i]) {
+            firstDiff = i;
+            break;
+        }
+    }
+    return firstDiff == -1;
+}
+
+const vector<string> &TestResourcesAndFunctions::getTestVectorWithSimulatedBlockData() {
+    lock_guard<mutex> lock(TestResourcesAndFunctions::staticLock);
+
+    if (!testVectorWithSimulatedDecompressedBlockData.empty())
+        return testVectorWithSimulatedDecompressedBlockData;
+
+    vector<string> *v = &testVectorWithSimulatedDecompressedBlockData;
+
+    v->emplace_back(readResourceFile("blockAndLineCalculations/block_0_complete"));
+    v->emplace_back(readResourceFile("blockAndLineCalculations/block_1_endopen"));
+    v->emplace_back(readResourceFile("blockAndLineCalculations/block_2_startendopen"));
+    v->emplace_back(readResourceFile("blockAndLineCalculations/block_5_empty"));
+    v->emplace_back(readResourceFile("blockAndLineCalculations/block_5_empty"));
+    v->emplace_back(readResourceFile("blockAndLineCalculations/block_3_startendopen"));
+    v->emplace_back(readResourceFile("blockAndLineCalculations/block_3.1_startendopen_nonewlines"));
```

Comment:
As everywhere, just concatenating lower-case words is really hard to read ...
Why not use camel-case also for these?